.github/workflows/book.yml

```diff
@@ -0,0 +1,35 @@
+name: deploy-book
+
+on: [push]
+
+# This job installs dependencies, builds the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    # Install dependencies
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install jupyterbook
+      run: |
+        pip install -r requirements.txt
+        pip install jupyter-book
+        python -m ipykernel install --user --name="cvx"
+        
+    # Build the book
+    - name: Build the book
+      run: |
+        jupyter-book clean book
```

Comment:
What does the `clean` step do?

Reply:
It deletes the html files etc. generated in previous iterations. They may not be there and probably this could be deleted in a ci process